xla/service/gpu/cudnn_norm_rewriter.cc

```diff
@@ -568,12 +589,13 @@ auto Expectation(UniqueHloInstruction* expectation, Pattern pattern) {
 template <typename Pattern>
 auto Expectation(UniqueHloInstruction* expectation, HloInstruction** reduce,
                  Pattern pattern) {
-  auto shared_subpattern = MultiplyAnyOrder(m::Broadcast(m::ConstantScalar()),
-                                            AddReduce(reduce, pattern))
-                               .WithPredicate([](const HloInstruction* instr) {
-                                 return CalculatesExpectation(instr);
-                               })
-                               .WithPredicate(expectation->capture_or_verify);
+  auto shared_subpattern = OptionalSupportedTransform(
```

Comment:
You add a lot of OptionalSupportedTransform calls. The PR description does state we now support "additional optional type conversions", but this also allows additional reshapes. The AdjustedDimensions function can take into account these reshapes, but I think it is not tested. Update the PR description to state additional reshapes removing/adding degenerate dimensions are allowed, and also add a test with such a reshape.